src/track/cueinfoimporter.cpp

```diff
@@ -0,0 +1,79 @@
+#include "track/cueinfoimporter.h"
+
+namespace mixxx {
+
+CueInfoImporter::CueInfoImporter() {
+}
+
+CueInfoImporter::CueInfoImporter(const QList<CueInfo>& cueInfos)
+        : m_cueInfos(cueInfos) {
+}
+
+double CueInfoImporter::getTimingOffsetMillis(
+        const QString& filePath,
+        const audio::SignalInfo& signalInfo) const {
+    Q_UNUSED(filePath);
+    Q_UNUSED(signalInfo);
+    return 0;
+};
+
+void CueInfoImporter::append(const CueInfo& cueInfo) {
+    m_cueInfos.append(cueInfo);
+}
+
+void CueInfoImporter::append(const QList<CueInfo>& cueInfos) {
+    m_cueInfos.append(cueInfos);
+}
+
+int CueInfoImporter::size() const {
+    return m_cueInfos.size();
+}
+
+bool CueInfoImporter::isEmpty() const {
+    return m_cueInfos.isEmpty();
+}
+
+QList<CueInfo> CueInfoImporter::getCueInfosWithCorrectTiming(
+        const QString& filePath,
+        const audio::SignalInfo& signalInfo) {
+    QList<CueInfo> cueInfos;
+
+    // Do not calculate offset if we don't have any cues to import
+    if (m_cueInfos.isEmpty()) {
+        cueInfos.append(m_cueInfos);
```

Comment:
`cueInfos = m_cueInfos`? Or maybe simply `return m_cueInfos`?

Reply:
@uklotzde requested that the CueInfoImporter object becomes empty after retrieving the cues, so we can't simply return that 
list: https://github.com/mixxxdj/mixxx/pull/2707#discussion_r414911370
The alternative would be to add a `clear()` method and emptying the internal list from outside. I'd also be okay with that.

Reply:
The intention is to perform this transformation exactly once. The code should reflect this use case and prevent any invalid usage if possible. This is achieved by consuming the input data during the conversion.

If you use a functional language with immutable data structures or Rust with its *move semantics* this pattern is very common ;)

Reply:
How about just `return cueInfos` with the default constructed `QList<CueInfo>`? Appending an empty list to an empty list doesn't do anything, does it?